app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,66 @@
+class CartsController < ApplicationController
+  before_action :authenticate_user!
+  rescue_from ActiveRecord::RecordNotFound, with: :invalid_cart
+  before_action :set_cart, only: [:show, :edit, :update, :destroy]
```

Comment:
change to `before_action :set_cart, only: %i[show edit update destroy]` and apply it for whole project